File: lizzie-mobile/shared/src/commonMain/kotlin/com/lizzie/rules/GameInfo.kt
package com.lizzie.rules

/**
 * Metadata about a Go game (players, komi, result, etc.)
 */
data class GameInfo(
    val playerBlack: String = "",
    val playerWhite: String = "",
    val komi: Double = 6.5,
    val handicap: Int = 0,
    val ruleSet: String = "jp",
    val gameName: String = "",
    val gameDate: String = "",
    val result: String = "",
) {
    fun isComplete(): Boolean = result.isNotEmpty()

    companion object {
        /** Parse a GTP komi response string to double. */
        fun parseKomi(value: String): Double = value.toDoubleOrNull() ?: 6.5
    }
}

File: lizzie_mobile/lib/go/game_info.dart
/// Game-level metadata (komi, handicap, player names).
///
/// Ported from Lizzie's `GameInfo.java`.
class GameInfo {
  final double komi;
  final int handicap;
  final String playerBlack;
  final String playerWhite;
  final String gameName;
  final String gameDate;
  final String result;

  const GameInfo({
    this.komi = 7.5,
    this.handicap = 0,
    this.playerBlack = '',
    this.playerWhite = '',
    this.gameName = '',
    this.gameDate = '',
    this.result = '',
  });

  GameInfo copyWith({
    double? komi,
    int? handicap,
    String? playerBlack,
    String? playerWhite,
    String? gameName,
    String? gameDate,
    String? result,
  }) {
    return GameInfo(
      komi: komi ?? this.komi,
      handicap: handicap ?? this.handicap,
      playerBlack: playerBlack ?? this.playerBlack,
      playerWhite: playerWhite ?? this.playerWhite,
      gameName: gameName ?? this.gameName,
      gameDate: gameDate ?? this.gameDate,
      result: result ?? this.result,
    );
  }
}
